tests/README.md

```diff
@@ -1,25 +1,239 @@
+```
+
+This will error if the Yocto environment has not been sourced yet.
+
+
+-------------------------------------------------------------------------------
+
+
+## Running the Tests in a VM
+
+### Vagrant
+
+The setup provides a
+[Vagrantfile](https://www.vagrantup.com/intro/getting-started/) in order to
+provide a common ground for running the integration tests on, based on an
+Ubuntu18.04 image. This can be found in `tests/vagrant`, and provisions a VM
+with the correct setup. The Vagrantfile will provision a fresh Ubuntu18.04
+image, and install all the *Debian* and *Python* requirements in the
+*requirements/(deb|python)requirements.txt* file.
+
+This requires that Vagrant is installed, as well as having the
+
+| Environment variable      | Reason |
+| ---                       | ---    |
+|GOPATH                     | Used for mounting your local GO tree in the VM as a synced folder |
+|DOCKER_USER                | Needed for logging in to the Mender docker registry  |
+|MENDER_DOCKER_REGISTRY_PWD |  Needed for logging in to the Mender docker registry  |
+
+set.
+
+Then in order to run the tests in the VM do
+
+```bash
+$ cd ./vagrant
+$ vagrant up
+$ vagrant ssh
+$ cd go/src/github.com/mendersoftware/integration/tests
+$ ./run.sh
+```
+
+
+# Legacy Stuff (TODO -- Remove)
+
+#### Requirements
 Python 2.7, requests, pytest, fabric, docker, docker-compose
 
-Install all Python requirements by running:  `pip2.7 install -r requirements.txt`
+Install all Python requirements by running: `pip2.7 install -r requirements.txt`
 
 Running tests is straight forward, just execute `bash run.sh`
 
-`run.sh` is simple wrapper that  grabs the latest test images, artifacts tools, and other test requirements. It also calls `py.test` with `--runfast` and `--runslow`, which runs all fast and slow tests.
+`run.sh` is simple wrapper that grabs the latest test images, artifacts tools,
+and other test requirements. It also calls `py.test` with `--runfast` and
+`--runslow`, which runs all fast and slow tests.
 
-####OS X
+#### OS X
 
-Currently, running integration tests on OS X is not straight forward due to: 
+Currently, running integration tests on OS X is not straight forward due to:
 https://github.com/docker/docker/issues/22753
 
-####Known issues
+#### Known issues
 
-Since we attempting to SSH into the virtual mender device, before the OS is up and running, you may see errors such as:
+Since we attempting to SSH into the virtual mender device, before the OS is up
+and running, you may see errors such as:
 
-`Fatal error: Needed to prompt for a connection or sudo password (host: 172.18.0.6:8822), but abort-on-prompts was set to True Aborting.`
+`Fatal error: Needed to prompt for a connection or sudo password (host:
+172.18.0.6:8822), but abort-on-prompts was set to True Aborting.`
 
 These can simply be ignored.
+
+## Tips and Tricks
+
+Remove the leftover `pycache` and `pyc` files, before testing.
```

Comment:
Why?

Reply:
Because they cache things like paths, so when running on Vagrant, you will get the wrong paths.

Reply:
Perhaps add this as Vagrant specific though. Yes. That I will.

Reply:
FWIW, I think you can set `PYTHONDONTWRITEBYTECODE` to avoid Python creating these files. ref  https://docs.python.org/3.8/using/cmdline.html#envvar-PYTHONDONTWRITEBYTECODE

Reply:
Nice, TIL.